csrc/preseg_passes/propagate_shardings.cpp

```diff
@@ -19,20 +19,13 @@ void PropagateShardingsPass::runPass(Fusion* fusion) {
   for (Expr* expr : fusion->exprs()) {
     const auto& inputs = ir_utils::filterByType<TensorView>(expr->inputs());
     const auto& outputs = ir_utils::filterByType<TensorView>(expr->outputs());
-    if (inputs.empty()) {
+
+    auto i = std::find_if(
```

Comment:
Single-device fusions don't have DeviceMeshes so the old code needs to be fixed. I moved the validation to the end.